core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -291,7 +290,7 @@ private[spark] class CoarseGrainedExecutorBackend(
       // is viewed as acceptable to minimize introduction of any new locking structures in critical
       // code paths.
 
-      val shutdownThread = new Thread() {
+      val shutdownThread = ThreadUtils.runInNewThread("wait-for-blocks-to-migrate") {
```

Comment:
I believe runInNewThread will do a "Thread.join" ... whereas you want this to be asynchronous here (atleast per the original implementation)

Reply:
ah yes good point.